course_info/static/course_info/js/controllers/PeopleController.js

```diff
@@ -234,103 +353,72 @@
-                                            role_id: $scope.selectedRole.roleId});
-                }
-                else {
-                    $scope.searchResults = filteredResults;
-                    $scope.searchInProgress = false;
-                }
-            }
+        $scope.handleAjaxErrorResponse = function(r) {
+            // angular promise then() function returns a response object,
+            // unpack for our old-style error handler
+            $scope.handleAjaxError(
+                r.data, r.status, r.headers, r.config, r.statusText);
         };
         $scope.isUnivID = function(searchTerm) {
             var re = /^[A-Za-z0-9]{8}$/;
             return re.test(searchTerm);
         };
-        $scope.lookup = function(searchTerm) {
-            var peopleParams = {page_size: 100};
-            var memberParams = {page_size: 100};
-
-            if ($scope.isUnivID(searchTerm)) {
-                peopleParams.univ_id = searchTerm;
-                memberParams.user_id = searchTerm;
-            } else {
-                peopleParams.email_address = searchTerm;
-                memberParams['profile.email_address'] = searchTerm;
-            }
-
-            // first the general people lookup
-            var peopleURL = djangoUrl.reverse('icommons_rest_api_proxy',
-                                              ['api/course/v2/people/']);
-            var peoplePromise = $http.get(peopleURL, {params: peopleParams,
-                                                      searchTerm: searchTerm})
-                                     .error($scope.handleAjaxError);
-
-            // then the course membership lookup
+        $scope.lookupCourseMembers = function() {
+            // exclude xreg people
+            var getConfig = {
+                params: {limit: 100, '-source': 'xreg_map'}};
             var memberURL = djangoUrl.reverse('icommons_rest_api_proxy',
                                               ['api/course/v2/course_instances/'
                                                + $scope.courseInstanceId
                                                + '/people/']);
-            var memberPromise = $http.get(memberURL, {params: memberParams,
-                                                      searchTerm: searchTerm})
-                                     .error($scope.handleAjaxError);
+            return angularDRF.get(memberURL, getConfig);
+        };
+        $scope.lookupPeople = function(searchTermList) {
+            /* generates a list of promises (http requests to the API) for the
+              search terms provided in order to get the univ_id for each
+              person the user wants to add
+              */
+            var peopleURL = djangoUrl.reverse('icommons_rest_api_proxy',
+                                              ['api/course/v2/people/']);
+            var peoplePromiseList = [];
+
+            searchTermList.forEach(function setupPersonLookup(searchTerm) {
```

Comment:
It occurs to me that this could just be

```javascript
return searchTermList.map(function setupPersonLookup(searchTerm) {
   ...
   return promise;
}
```

Rather than creating and appending to an array.  Dunno if that's any more readable, though.